### PR TITLE
Protocol sanity check

### PR DESCRIPTION
**Pull request by 409H**
* Checks the protocol on the link added to 'continue at own risk'. If it is an unsupported protocol then show a message to the user

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -121,6 +135,11 @@ function start() {
   }
 
   continueLink.addEventListener('click', async () => {
+    if (isValidSuspectHref(suspectHref) === false) {
+      console.log(`Disallowed Protocol, cannot continue.`);
```
Comment (adonesky1) [inline src/index.ts:139]: Will this appear to fail silently to a user?

Comment (409H) [reply src/index.ts:139]: It will fail silently to the user. However, typically, a user will be directed to this view after trying to access an endpoint via `^https?:\/\/`. 

If we have a UX recommendation so that we can not fail silently to the user, I am all ears :D

Comment (adonesky1) [reply src/index.ts:139]: Sorry I'm a bit confused. What will the user see if they hit this condition? -> The phishing page stays put, and as they click the `continue at your own risk` button, they see no indication as to why the button appears to not work?

File: src/index.test.ts [hunk 2]
```diff
@@ -103,6 +103,10 @@ describe('Phishing warning page', () => {
     'should redirect to the site after the user continues at their own risk',
   );
 
+  it.todo(
```
Comment (mcmire) [inline src/index.test.ts:106]: Can you add a test that provides an example for why we wouldn't want to redirect?

File: src/index.ts (hunk 1 above)
Comment (mcmire) [reply src/index.ts:139]: Piggybacking on Alex's request a bit, what's the context that prompted this change? Why would we not want to redirect in this case? I'm also wondering if perhaps in this case instead of having a button that doesn't do anything, perhaps we should use some different copy that explains what's special about this particular URL and then we just _don't_ show a button.

Comment (adonesky1) [reply src/index.ts:139]: > I'm also wondering if perhaps in this case instead of having a button that doesn't do anything, perhaps we should use some different copy that explains what's special about this particular URL and then we just don't show a button.

Yea this is what I'm thinking as well

Comment (adonesky1) [reply src/index.ts:149]: LGTM

File: src/index.test.ts [hunk 3]
```diff
@@ -63,7 +63,7 @@ describe('Phishing warning page', () => {
   });
 
   afterEach(() => {
-    onDomContentLoad = undefined;
```
Comment (adonesky1) [inline src/index.test.ts:66]: Why is this change required?

Comment (409H) [reply src/index.test.ts:66]: Idk, but the test and content only resolves as it should with it removed ¯\_(ツ)_/¯

**Review by adonesky1 — APPROVED** [review]
LGTM
